File: nvxapp.client/src/nvx/ClientServer-Service/GestionePresenze/Par_Orario/Models/par-orario-model.ts
import { ModelResult } from "../../../ModelsBase/model-result";
import { Par_OrarioIntervalloHHModel } from "../../Par_OrarioIntervalloHH/Models/par-orario-intervallo-hh-model";

export class Par_OrarioModel {
  id: number;
  idAz_Anagrafica: number;
  codice: string;
  descrizione: string;
  numeroCoppie: number;
  sogliaHHStrao: number;
}

export class Par_Orario_GetAllInModel { }
export class Par_Orario_GetAllOutModel extends ModelResult {
  par_Orario: Par_OrarioModel[];
  constructor() {
    super();
    this.par_Orario = [];
  }
}




export class Par_Orario_GetInModel {
  id: number;
}
export class Par_Orario_GetOutModel extends ModelResult {
  par_Orario: Par_OrarioModel;
  par_OrarioIntervalloHH: Par_OrarioIntervalloHHModel[];
  constructor() {
    super();
    this.par_Orario = new Par_OrarioModel();
    this.par_OrarioIntervalloHH = [];
  }
}

export class Par_Orario_PutInModel {
  par_Orario: Par_OrarioModel;
  par_OrarioIntervalloHH: Par_OrarioIntervalloHHModel[];
  constructor() {
    this.par_Orario = new Par_OrarioModel();
    this.par_OrarioIntervalloHH = [];
  }
}
export class Par_Orario_PutOutModel extends ModelResult {
  par_Orario: Par_OrarioModel;
  par_OrarioIntervalloHH: Par_OrarioIntervalloHHModel[];
  constructor() {
    super();
    this.par_Orario = new Par_OrarioModel();
    this.par_OrarioIntervalloHH = [];
  }
}

export class Par_Orario_DeleteInModel {
  id: number;
}
export class Par_Orario_DeleteOutModel extends ModelResult { }

File: nvxapp.server.service/ClientServer-Service/GestionePresenze/Par_OrarioService/Models/Par_OrarioModel.cs
using nvxapp.server.service.ClientServer_Service.ModelsBase;
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace nvxapp.server.service.ClientServer_Service.GestionePresenze.Par_OrarioService.Models
{
    public class Par_OrarioModel
    {
        public int Id { get; set; }
        public int IdAz_Anagrafica { get; set; }
        public string Codice { get; set; } = string.Empty;
        public string Descrizione { get; set; } = string.Empty;
        public int NumGiorniCiclo { get; set; }
        
    }

    public class Par_Orario_GetAllInModel { }
    public class Par_Orario_GetAllOutModel : ModelResult
    {
        public List<Par_OrarioModel> Par_Orario { get; set; } = new List<Par_OrarioModel>();
    }

    public class Par_Orario_GetInModel
    {
        public int Id { get; set; }
    }
    public class Par_Orario_GetOutModel : ModelResult
    {
        public Par_OrarioModel? Par_Orario { get; set; }
    }

    public class Par_Orario_PutInModel
    {
        public Par_OrarioModel Par_Orario { get; set; } = new Par_OrarioModel();
    }
    public class Par_Orario_PutOutModel : ModelResult
    {
        public Par_OrarioModel? Par_Orario { get; set; }
    }

    public class Par_Orario_DeleteInModel
    {
        public int Id { get; set; }
    }
    public class Par_Orario_DeleteOutModel : ModelResult { }
}
